### PR TITLE
[com_menu] - delete reference on #_modules_menu deleting a menu item

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #13108 .

### Summary of Changes
delete reference on `#_modules_menu` table when is deleted a trashed menu item

### Testing Instructions

see https://github.com/joomla/joomla-cms/issues/13108#issue-193896370

### CODE REVIEW

**Comment by ralain** [comment]
I have tested this item :red_circle: unsuccessfully on 91d1c062cf40fd65528431a633f24febb2e3c20b<br /><br />Issue remains. Checked database and the reference is not removed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13126">issues.joomla.org/tracker/joomla-cms/13126</a>.</sub>

File: administrator/components/com_menus/tables/menu.php [hunk 1]
```diff
@@ -28,6 +28,19 @@ class MenusTableMenu extends JTableMenu
...
-		return parent::delete($pk, $children);
+		$return = parent::delete($pk, $children);
+
+		if ($return)
+		{
+			// Delete key from the #__modules_menu table
+			$db = JFactory::getDbo();
+			$query = $db->getQuery(true)
+				->delete($db->quoteName('#__modules_menu'))
+				->where($db->quoteName('moduleid') . ' = ' . $pk);
```
Comment (ralain) [inline administrator/components/com_menus/tables/menu.php]: Shouldn't we be looking for the menuid?

Comment (alikon) [inline administrator/components/com_menus/tables/menu.php]: yes thanks for reporting

**Comment by ralain** [comment]
I have tested this item :white_check_mark: successfully on 84d656521b54de6f851ebb75f6c535b2f5f67959<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13126">issues.joomla.org/tracker/joomla-cms/13126</a>.</sub>

**Comment by coolcat-creations** [comment]
I have tested this item :white_check_mark: successfully on 84d656521b54de6f851ebb75f6c535b2f5f67959<br /><br />thanks :) works now <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13126">issues.joomla.org/tracker/joomla-cms/13126</a>.</sub>

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13126">issues.joomla.org/tracker/joomla-cms/13126</a>.</sub>

**Comment by wilsonge** [comment]
Good job :)

**Pull request merged**